Example "image-classification-web" does not build

**Describe the bug**
Trying to build the example **image-classification-web** either from `main` branch or from `release/0.13` fails with the following error:
```bash
error[E0599]: no variant or associated item named `default` found for enum `kernel::matmul::base::MatmulStrategy` in the current scope
   --> crates/burn-jit/src/ops/float_ops.rs:149:42
    |
149 |         matmul(lhs, rhs, MatmulStrategy::default())
    |                                          ^^^^^^^ variant or associated item not found in `MatmulStrategy`
    |
   ::: crates/burn-jit/src/kernel/matmul/base.rs:11:1
    |
11  | pub enum MatmulStrategy {
    | ----------------------- variant or associated item `default` not found for this enum
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `default`, perhaps you need to implement it:
            candidate #1: `std::default::Default`
```

**To Reproduce**
 Steps to reproduce the behavior:
 1. Go to _**'examples/image-classification-web'**_
 2. execute _**./build-for-web.sh**_

--> See the build process fail with the error mentioned above.

**Expected behavior**
Successfull build

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
 - rustc: 1.79.0
 - cargo: 1.79.0

**Additional context**
There is a `default()` defined in feature `autotune` as 
```rust
#[cfg(not(feature = "autotune"))]
impl Default for MatmulStrategy {
    fn default() -> Self {
        MatmulStrategy::Tiling2d(Tiling2dConfig::default())
    }
}
```
However, the `autotune` feature as well as other `default` features do not seem to be activated. This may be due to `core-jit`s membership in the **workspace**.

--- a/crates/burn-jit/src/kernel/matmul/base.rs
+++ b/crates/burn-jit/src/kernel/matmul/base.rs
@@ -3,9 +3,12 @@ use burn_cube::prelude::*;
 use burn_tensor::Shape;
 
 use super::{
-    config::Tiling2dConfig, init_matmul_output, matmul_autotune, matmul_simple, matmul_tiling_2d,
+    config::Tiling2dConfig, init_matmul_output, matmul_simple, matmul_tiling_2d,
     matmul_tiling_2d_cube, matmul_tiling_2d_padded,
 };
+
+#[cfg(feature = "autotune")]
+use super::matmul_autotune;
 
 /// The strategy to be used when launching a matmul kernel.
 pub enum MatmulStrategy {
@@ -27,11 +30,14 @@ pub enum MatmulStrategy {
     Tiling2dCube(Tiling2dConfig),
 }
 
-#[allow(clippy::derivable_impls)] // Necessary otherwise the feature flags dont' work.
-#[cfg(feature = "autotune")]
 impl Default for MatmulStrategy {
     fn default() -> Self {
-        MatmulStrategy::Autotune
+        // if autotune is enabled, default to autotune
+        #[cfg(feature = "autotune")]
+        return MatmulStrategy::Autotune;
+
+        #[cfg(not(feature = "autotune"))]
+        MatmulStrategy::Tiling2d(Tiling2dConfig::default())
     }
 }
 

--- a/crates/burn-jit/src/kernel/matmul/tune/mod.rs
+++ b/crates/burn-jit/src/kernel/matmul/tune/mod.rs
@@ -1,5 +1,7 @@
+#[cfg(feature = "autotune")]
 mod base;
 mod key;
 
+#[cfg(feature = "autotune")]
 pub use base::*;
 pub use key::*;

--- a/crates/burn-jit/src/kernel/reduce/base.rs
+++ b/crates/burn-jit/src/kernel/reduce/base.rs
@@ -7,6 +7,7 @@ use super::{
     shared::{base::ReduceDimShared, shader::reduce_dim_shared},
 };
 
+#[allow(dead_code)]
 pub(crate) trait ReduceDimAlgorithm<E: JitElement>:
     ReduceDimNaive<E> + ReduceDimShared<E>
 {

--- a/crates/burn-jit/src/kernel/reduce/tune/mod.rs
+++ b/crates/burn-jit/src/kernel/reduce/tune/mod.rs
@@ -1,5 +1,7 @@
+#[cfg(feature = "autotune")]
 mod base;
 mod key;
 
+#[cfg(feature = "autotune")]
 pub(crate) use base::*;
 pub use key::*;

--- a/examples/image-classification-web/Cargo.toml
+++ b/examples/image-classification-web/Cargo.toml
@@ -17,7 +17,9 @@ half_precision = []
 burn = { path = "../../crates/burn", version = "0.14.0", default-features = false, features = [
     "ndarray",
 ] }
-burn-wgpu = { path = "../../crates/burn-wgpu", version = "0.14.0", default-features = false }
+burn-wgpu = { path = "../../crates/burn-wgpu", version = "0.14.0", default-features = false, features = [
+    "autotune",
+] }
 burn-candle = { path = "../../crates/burn-candle", version = "0.14.0", default-features = false }
 
 log = { workspace = true }

